### ISSUE TITLE
Parse usage section

### ISSUE THREAD

**Issue by hadley** [r-lib/pkgdown#288]
Enables:

* documentation & source lookup (#234)
* better listing on reference index (#256)

```
name(args)
x %name% y
name(args) <- values

\method{generic}{class}(args)
\method{[}{factor}(args)
\method{generic}{class}(arglist) <- value
```

`\S3method{}` is an alias for method; `\S4method{}` is used for S4 methods.

To parse, a reasonable strategy would be to replace `\S3method{x}{y}` with ``S3method(`x`, list(y))`` and then rely on `parse()`. Will have to do a little post processing if discover assignment. Then need to look up in the package namespace, using `get()`, `getS3method()`, and `getMethod()`.

**Comment by hadley** [comment]
This could also enable a richer html structure in the usage pre statement, allowing for finer control over the spacing between entries.

**Comment by jayhesselberth** [comment]
This was inadvertently closed in #322. You should re-open, they are unrelated.
